jsapp/js/account/plan.component.tsx

```diff
@@ -0,0 +1,339 @@
+    case 'month':
+      return {...state, intervalFilter: 'month', filterToggle: !state.filterToggle};
+    case 'year':
+      return {...state, intervalFilter: 'year', filterToggle: !state.filterToggle};
+    default:
+      return state;
+  }
+}
+
+export default function Plan() {
+  const [state, dispatch] = useReducer(planReducer, initialState);
+  const [expandComparison, setExpandComparison] = useState(false);
+  const [buttonsDisabled, setButtonDisabled] = useState(false);
+
+  useEffect(() => {
+    getProducts().then((data) => {
+      dispatch({
+        type: 'initialProd', prodData: data.results,
+      });
+    });
+
+    getOrganization().then((data) => {
+      console.log('org', data);
+
+      dispatch({
+        type: 'initialOrg', prodData: data.results[0],
+      });
+    });
+
+    getSubscription().then((data) => {
+      console.log('sub', data);
+      dispatch({
+        type: 'initialSub', prodData: data.results,
+      });
+    });
+  }, []);
+
+  // Filter prices based on plan interval
+  const filterPrices = () => {
+    if (state.products.length > 0) {
+      const filterAmount = state.products.map((product: any) => {
+        const filteredPrices = product.prices.filter((price: any) => {
+          const interval = price.human_readable_price.split('/')[1];
+          return interval === state.intervalFilter || price.unit_amount === 0;
+        });
+        return {
+          ...product,
+          prices: filteredPrices.length ? filteredPrices[0] : null,
+        };
+      });
+      return filterAmount.filter((product: any) => product.prices);
+    }
+  };
+
+  const isSubscribedProduct = (product: any) => {
+    return product.name === state.subscribedProduct?.name;
+  };
+
+  const upgradePlan = (priceId: string) => {
+    console.log('Upgrade', priceId);
+    if (!priceId || buttonsDisabled){
+      return;
+    }
+    setButtonDisabled(buttonsDisabled);
+    $.ajax({
+      dataType: 'json',
+      method: 'POST',
+      url: `${ROOT_URL}/api/v2/stripe/checkout-link?price_id=${priceId}&organization_uid=${state.organization?.uid}`,
+    })
+      .done(function (res) {
+        window.location.assign(res.url);
+      })
+      .fail(handleApiFail)
+      .always(() =>
+        setButtonDisabled(!buttonsDisabled)
+      );
+  };
+
+  const managePlan = () => {
+    console.log('Manage', state.organization?.uid);
+    if (!state.organization?.uid || buttonsDisabled){
+      return;
+    }
+    setButtonDisabled(buttonsDisabled);
+    $.ajax({
+      dataType: 'json',
+      method: 'POST',
+      url: `${ROOT_URL}/api/v2/stripe/customer-portal?organization_uid=${state.organization?.uid}`,
+    })
+      .done(function (res) {
+        window.location.assign(res.url);
+      })
+      .fail(handleApiFail)
+      .always(() =>
+        setButtonDisabled(!buttonsDisabled)
+      );
+  };
+
+  // Get feature items and matching icon boolean
+  const getListItem = (listType: string, plan: string) => {
```

Comment:
We discussed previously that I have a slight lean towards using useMemo more often for consistency and why not. But it's true that it probably won't ever matter how many "extra" times this JS is executed. I'll make the call to say that this is not a merge blocker but happy to keep the conversation going.

Reply:
I am open to moving it to a "useMemo" - my only thought against was to follow best practice for only using when it can improve performance since it could do the opposite if we standardize adding it. Honestly I find it hard to tell with some of these smaller functions when it is useful and when its not